src/settings.h

```diff
@@ -71,6 +71,24 @@ class Settings {
     showThumbnails_ = show;
   }
 
+  QString thumbnailsPosition() {
+    return thumbnailsPosition_;
+  }
+  void setThumbnailsPosition(QString position) {
+    thumbnailsPosition_ = position;
+  }
+  const QMap<QString, Qt::DockWidgetArea>& thumbnailsPositions() {
+    static const QMap<QString, Qt::DockWidgetArea> _thumbnailsPositions = {
+        {QStringLiteral("left"), Qt::LeftDockWidgetArea},
+        {QStringLiteral("right"), Qt::RightDockWidgetArea},
+        {QStringLiteral("top"), Qt::TopDockWidgetArea},
+        {QStringLiteral("bottom"), Qt::BottomDockWidgetArea} };
+    return _thumbnailsPositions;
+  }
+  Qt::DockWidgetArea getThumbnailsPosition() {
+    return thumbnailsPositions()[thumbnailsPosition_];
+  }
```

Comment:
`QMap` isn't needed here:

```c++
  QString thumbnailsPosition() {
    return thumbnailsPosition_;
  }
  void setThumbnailsPosition(const QString& position) {
    if(!thumbnailsPositions().contains(position)) {
      thumbnailsPosition_ = QStringLiteral("bottom");
    }
    else {
      thumbnailsPosition_ = position;
    }
  }
  const QStringList& thumbnailsPositions() {
    static const QStringList _thumbnailsPositions = {QStringLiteral("left"),
                                                     QStringLiteral("right"),
                                                     QStringLiteral("top"),
                                                     QStringLiteral("bottom")};
    return _thumbnailsPositions;
  }
  Qt::DockWidgetArea getThumbnailsPosition() {
    if(thumbnailsPosition_ == QStringLiteral("left")) {
      return Qt::LeftDockWidgetArea;
    }
    if(thumbnailsPosition_ == QStringLiteral("right")) {
      return Qt::RightDockWidgetArea;
    }
    if(thumbnailsPosition_ == QStringLiteral("top")) {
      return Qt::TopDockWidgetArea;
    }
    return Qt::BottomDockWidgetArea;
  }
```

It also guards against meaningless strings in the config file (which can be edited manually).

It's preferable to put function definitions in `settings.cpp`.

Reply:
I'll make these changes.

I put these function definitions here because every other function of this type in this section is defined here.

Reply:
> I put these function definitions here because...

I have no objection.